services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java

```diff
@@ -165,88 +177,227 @@ public String getId() {
+                    // Create the default HTTP client with no retries
+                    httpClient = createHttpClientWithoutRetries(session);
 
-                        logger.debugf("httpProxy: %s, noProxy: %s", httpProxy, noProxy);
-                        proxyMappings = ProxyMappings.withFixedProxyMapping(httpProxy, noProxy);
+                    // Initialize the default retriable client
+                    if (defaultRetryConfig.getMaxRetries() > 0) {
+                        // Create a retriable client with the default configuration
+                        retriableHttpClient = createRetriableHttpClient(session, defaultRetryConfig);
+                    } else {
+                        // If retries are disabled by default, use the regular client
+                        retriableHttpClient = httpClient;
                     }
+                }
+            }
+        }
+    }
 
-                    HttpClientBuilder builder = newHttpClientBuilder();
-
-                    builder.socketTimeout(socketTimeout, TimeUnit.MILLISECONDS)
-                            .establishConnectionTimeout(establishConnectionTimeout, TimeUnit.MILLISECONDS)
-                            .maxPooledPerRoute(maxPooledPerRoute)
-                            .connectionPoolSize(connectionPoolSize)
-                            .connectionTTL(connectionTTL, TimeUnit.MILLISECONDS)
-                            .maxConnectionIdleTime(maxConnectionIdleTime, TimeUnit.MILLISECONDS)
-                            .disableCookies(disableCookies)
-                            .proxyMappings(proxyMappings)
-                            .expectContinueEnabled(expectContinueEnabled)
-                            .reuseConnections(reuseConnections);
-
-                    TruststoreProvider truststoreProvider = session.getProvider(TruststoreProvider.class);
-                    boolean disableTruststoreProvider = truststoreProvider == null || truststoreProvider.getTruststore() == null;
+    /**
+     * Creates a retriable HTTP client with the specified retry configuration
+     */
+    private CloseableHttpClient createRetriableHttpClient(KeycloakSession session, RetryConfig retryConfig) {
+        // If retries are disabled, just return the default client
+        if (retryConfig == null || retryConfig.getMaxRetries() <= 0) {
+            return httpClient;
+        }
 
-                    if (disableTruststoreProvider) {
-                    	logger.warn("TruststoreProvider is disabled");
-                    } else {
-                        builder.hostnameVerification(truststoreProvider.getPolicy());
-                        try {
-                            builder.trustStore(truststoreProvider.getTruststore());
-                        } catch (Exception e) {
-                            throw new RuntimeException("Failed to load truststore", e);
+        // Create HTTP client builder
+        HttpClientBuilder builder = newHttpClientBuilder();
+
```

Comment:
I wonder if we could revert the implementation to the original on, extract these settings into a separate method and call it like this:

```
RetryConfig retryConfig = ... ;
builder.retries(retryConfig);
```

All the wiring would go to the builder and the implementation would be very small and lean in this class. 

What do you think about this approach?